Crash report 3.9: NPE in WPWebViewClient.shouldInterceptRequest

```
java.lang.ExceptionInInitializerError
       at android.webkit.CallbackProxy.shouldOverrideUrlLoading(CallbackProxy.java:1123)
       at android.webkit.BrowserFrame.handleUrl(BrowserFrame.java:648)
       at android.webkit.WebViewCore.nativeMouseClick(WebViewCore.java)
       at android.webkit.WebViewCore.access$7700(WebViewCore.java:63)
       at android.webkit.WebViewCore$EventHub.dispatchWebKitEvent(WebViewCore.java:1996)
       at android.webkit.WebViewInputDispatcher.dispatchWebKitEvent(WebViewInputDispatcher.java:706)
       at android.webkit.WebViewInputDispatcher.dispatchWebKitEvents(WebViewInputDispatcher.java:641)
       at android.webkit.WebViewInputDispatcher.access$900(WebViewInputDispatcher.java:78)
       at android.webkit.WebViewInputDispatcher$WebKitHandler.handleMessage(WebViewInputDispatcher.java:1170)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:194)
       at android.webkit.WebViewCore$WebCoreThread.run(WebViewCore.java:930)
       at java.lang.Thread.run(Thread.java:838)
Caused by: java.lang.NullPointerException
       at org.wordpress.android.util.WPWebViewClient.shouldInterceptRequest(WPWebViewClient.java:104)
       at android.webkit.CallbackProxy.shouldInterceptRequest(CallbackProxy.java:1203)
       at android.webkit.BrowserFrame.shouldInterceptRequest(BrowserFrame.java:876)
       at android.webkit.WebViewCore.nativeMouseClick(WebViewCore.java)
       at android.webkit.WebViewCore.access$7700(WebViewCore.java:63)
       at android.webkit.WebViewCore$EventHub.dispatchWebKitEvent(WebViewCore.java:1996)
       at android.webkit.WebViewInputDispatcher.dispatchWebKitEvent(WebViewInputDispatcher.java:706)
       at android.webkit.WebViewInputDispatcher.dispatchWebKitEvents(WebViewInputDispatcher.java:641)
       at android.webkit.WebViewInputDispatcher.access$900(WebViewInputDispatcher.java:78)
       at android.webkit.WebViewInputDispatcher$WebKitHandler.handleMessage(WebViewInputDispatcher.java:1170)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:194)
       at android.webkit.WebViewCore$WebCoreThread.run(WebViewCore.java:930)
       at java.lang.Thread.run(Thread.java:838)
```

--- a/WordPress/src/main/java/org/wordpress/android/util/WPWebViewClient.java
+++ b/WordPress/src/main/java/org/wordpress/android/util/WPWebViewClient.java
@@ -99,6 +99,8 @@ public class WPWebViewClient extends WebViewClient {
                 URL url = new URL(stringUrl);
                 HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
                 urlConnection.setRequestProperty("Authorization", "Bearer " + mToken);
+                urlConnection.setReadTimeout(WPRestClient.REST_TIMEOUT_MS);
+                urlConnection.setConnectTimeout(WPRestClient.REST_TIMEOUT_MS);
                 WebResourceResponse response = new WebResourceResponse(urlConnection.getContentType(),
                         urlConnection.getContentEncoding(),
                         urlConnection.getInputStream());

--- a/WordPress/src/main/java/org/wordpress/android/util/WPWebViewClient.java
+++ b/WordPress/src/main/java/org/wordpress/android/util/WPWebViewClient.java
@@ -12,15 +12,14 @@ import android.webkit.WebResourceResponse;
 import android.webkit.WebView;
 import android.webkit.WebViewClient;
 
-import org.apache.http.HttpResponse;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.impl.client.DefaultHttpClient;
 import org.wordpress.android.WordPress;
 import org.wordpress.android.models.Blog;
 import org.wordpress.android.networking.SelfSignedSSLCertsManager;
 
 import java.io.IOException;
-import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.security.GeneralSecurityException;
 
 /**
@@ -93,22 +92,25 @@ public class WPWebViewClient extends WebViewClient {
     }
 
     @Override
-    public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
+    public WebResourceResponse shouldInterceptRequest(WebView view, String stringUrl) {
         // Intercept requests for private images and add the WP.com authorization header
-        if (mBlog != null && mBlog.isPrivate() && !TextUtils.isEmpty(mToken) && UrlUtils.isImageUrl(url)) {
-            DefaultHttpClient client = new DefaultHttpClient();
-            HttpGet httpGet = new HttpGet(url);
-            httpGet.setHeader("Authorization", "Bearer " + mToken);
+        if (mBlog != null && mBlog.isPrivate() && !TextUtils.isEmpty(mToken) && UrlUtils.isImageUrl(stringUrl)) {
             try {
-                HttpResponse httpResponse = client.execute(httpGet);
-                InputStream responseInputStream = httpResponse.getEntity().getContent();
-                return new WebResourceResponse(httpResponse.getEntity().getContentType().toString(),
-                        "UTF-8", responseInputStream);
+                URL url = new URL(stringUrl);
+                HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+                urlConnection.setRequestProperty("Authorization", "Bearer " + mToken);
+                WebResourceResponse response = new WebResourceResponse(urlConnection.getContentType(),
+                        urlConnection.getContentEncoding(),
+                        urlConnection.getInputStream());
+                return response;
+            } catch (ClassCastException e) {
+                AppLog.e(AppLog.T.POSTS, "Invalid connection type - URL: " + stringUrl);
+            } catch (MalformedURLException e) {
+                AppLog.e(AppLog.T.POSTS, "Malformed URL: " + stringUrl);
             } catch (IOException e) {
                 AppLog.e(AppLog.T.POSTS, "Invalid post detail request: " + e.getMessage());
             }
         }
-
-        return super.shouldInterceptRequest(view, url);
+        return super.shouldInterceptRequest(view, stringUrl);
     }
 }

